Gateway/Request/APMBuilder.php

```diff
@@ -280,11 +286,24 @@ public function build(array $buildSubject)
                 break;
             case Shopeepay::CODE:
                 $paymentInfo[self::SOURCE] = [
-                    self::SOURCE_TYPE => 'shopeepay',
+                    self::SOURCE_TYPE => $this->getShopeepaySource()
                 ];
                 break;
         }
 
         return $paymentInfo;
     }
+
+    private function getShopeepaySource()
+    {
+        $defaultSourceType = Shopeepay::ID;
+        $isShopeepayJumpAppEnabled = $this->capabilities->isBackendEnabled(Shopeepay::JUMPAPP_ID);
+        $isShopeepayEnabled = $this->capabilities->isBackendEnabled(Shopeepay::ID);
+
+        if ($this->helper->isMobilePlatform() && $isShopeepayJumpAppEnabled) {
+            return Shopeepay::JUMPAPP_ID;
+        }
+
+        return $isShopeepayEnabled ? Shopeepay::ID : Shopeepay::JUMPAPP_ID;
```

Comment:
what the use for `$defaultSourceType` na maybe we want default to be `shopeepay` right? 🤔

Reply:
Ah that variable won't be needed. I forgot to remove that line. `Shopeepay::ID` value is `shopeepay`. So, we are returning `shopeepay` if it is enabled. Before that we are checking for mobile device.  I have added comments to make it clear.